binary/capabilities.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package io.substrait;
+
+option java_multiple_files = true;
+option csharp_namespace = "Substrait.Protobuf";
+
+// Defines a set of Capabilities that a system (producer or consumer) supports.
+message Capabilities {
+
+    // List of Substrait versions this system supports
+    repeated string substrait_versions = 1;
+
+    // list of com.google.Any message types this system supports for advanced extensions.
+    repeated string advanced_extension_type_urls = 2;
+
+    // list of simple extensions this system supports.
+    repeated SimpleExtension simple_extensions = 3;
+
+    message SimpleExtension {
```

Comment:
Can you split the advanced/simple extension changes into a separate PR?

Reply:
done